sdk/python/approzium/_iam.py

```diff
@@ -36,3 +40,21 @@ def get_local_arn():
     sts_client = boto3.client("sts")
     response = sts_client.get_caller_identity()
     return response["Arn"]
+
+
+@lru_cache()
```

Comment:
Couple of questions:

- Should we set a max size here so the cache doesn't grow unboundedly?
- Can you link me to an example of the response we'd get from `"http://169.254.169.254/latest/dynamic/instance-identity/"`? I'm looking at an example of a _document_ [here](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-identity-documents.html) and I'm not sure if it'd be the same. 
- What would be the data that can change from call to call?
- Why a Least Recently Used cache specifically?
- If someone tried to speed up their application by having multiple clients running using something like the [multiprocessing](https://docs.python.org/3/library/multiprocessing.html) package, how would this `lru_cache` work? Is it thread-safe? I didn't see any mention in [the docs](https://docs.python.org/3/library/functools.html#functools.lru_cache).

Thanks! Just trying to get onto the same page around this.

Reply:
Great questions!
- The LRU cache simply calculates the response on the first call and returns the same result on all subsequent calls. The LRU cache tries to remember the result for each argument value, but since we don't have any arguments, it can only store one result (empty invocation of function) and nothing more. So, I don't think setting a max size is necessary.
- I didn't look into what it contains; my reasoning was that, if this address does return something, then it is probably an EC2 instance. Should we try to check the contents to make the check more robust?
- No change from call to call, I just wanted this check to happen only once (on the first attribution_info call) and keep returning the result from then on, because we don't expect a non-EC2 instance to change into an EC2-instance or vice versa.
- Not sure about this one 🤔 It looks like it is not thread safe by default. But I am not sure if the AuthClient itself is thread safe, or how I would reason about that. I am very comfortable with thread-safety in C but I am lacking in Python.

I think the thread-safety issue is interesting! We could explore it and have some tests/examples that demonstrate it. Should I create an issue?

Reply:
Hey! Ah, thanks for explaining!

Well, question then, how would you feel about setting a cache max size to 1? Just in case. That way if something changes, we don't end up debugging weird out-of-memory issues on the client side that are associated with this cache, because they could be hard to find.

Regarding checking the content of the document for changes, I think we don't have to do that. I didn't phrase my question very well, I was mainly looking for an example response to the call so I would understand if fields ever change. It sounds like they don't, so we're cool.

Regarding thread-safety, since we don't expect the result to ever change, we're okay as-is. If it's a write-once, read-forever situation, we don't have to worry about races. Immutable objects are inherently thread-safe.